include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp

```diff
@@ -1871,6 +1911,112 @@ struct MinMaxConverter : public OpRewritePattern<CmpOp> {
     return success();
   }
 
+  /// foldCmpSelectToMinMax performs an optimization pattern that matches
+  /// 'triton::ReduceOp' operations whose reduction body consists of a
+  /// single 'arith.select' operation based on a floating-point comparsion,
+  /// and rewrites them into equivalent 'arith.minf', 'arith.maxf',
+  /// 'arith.MinimumFOp' or 'arith.MaximumFOp' operations.
+  ///
+  /// This pattern handles the following cases:
+  ///
+  /// 1. ** Simple Min/Max Reduction **
+  ///    - select (cmpf ogt a, b), a, b --> maxf(a, b)
+  ///    - select (cmpf olt a, b), a, b --> minf(a, b)
+  ///
+  /// 2. ** NaN-Aware Min/Max Reduction **
+  ///    - select (cmpf ogt a, b) || cmpf une a, a), a, b --> arith.maximumf(a,
+  ///    b)
+  ///    - select (cmpf olt a, b) || cmpf une a, a), a, b --> arith.minimumf(a,
+  ///    b)
+  ///
+  /// These transformations not only improve IR canonicalization but also
+  /// allow the successful lowering of tt.reduce operations to linalg
+  /// operations, which is already supported in the triton-shared dialect
+  /// conversion pipeline.
+
+  LogicalResult foldCmpSelectToMinMax(PatternRewriter &rewriter,
```

Comment:
i think this function has a lot of similarities with the existing code in this pattern. could we try to consolidate them?